example/pages/index.tsx

```diff
@@ -31,13 +37,57 @@ const jsontheme = {
   base0F: '#a3685a',
 }
 
+function useLocalStorage<T extends any>(key: string, initialValue: T) {
+  // State to store our value
+  // Pass initial state function to useState so logic is only executed once
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    if (typeof window === 'undefined') {
+      return initialValue
+    }
+    try {
+      // Get from local storage by key
+      const item = window.localStorage.getItem(key)
+      // Parse stored json or if none return initialValue
+      return item ? JSON.parse(item) : initialValue
+    } catch (error) {
+      // If error also return initialValue
+      console.log(error)
+      return initialValue
+    }
+  })
+
+  // Return a wrapped version of useState's setter function that ...
+  // ... persists the new value to localStorage.
+  const setValue = (value: T): void => {
+    try {
+      // Allow value to be a function so we have same API as useState
+      const valueToStore =
+        value instanceof Function ? value(storedValue) : value
+      // Save state
+      setStoredValue(valueToStore)
+      // Save to local storage
+      if (typeof window !== 'undefined') {
+        window.localStorage.setItem(key, JSON.stringify(valueToStore))
+      }
+    } catch (error) {
+      // A more advanced implementation would handle the error case
+      console.log(error)
+    }
+  }
+  return [storedValue, setValue] as const
+}
+
 export default function Home(): React.ReactElement {
   const [analytics, setAnalytics] = useState<Analytics | undefined>(undefined)
-  const [settings, setSettings] = useState<AnalyticsSettings | undefined>(
-    undefined
-  )
+  const [writeKey, setWriteKey] = useLocalStorage<string>('__dev_writeKey', '')
+  const [settings, setSettings] = useState<
+    AnalyticsBrowserSettings | undefined
+  >({
+    writeKey,
+    cdnURL: 'https://cdn.segment.com',
+  })
```

Comment:
```suggestion
  const [writeKey, setWriteKey] = useLocalStorage<string>('__dev_writeKey', '')
  const [settings, setSettings] = useState<
    AnalyticsBrowserSettings | undefined
  >({
    writeKey: process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY,
    cdnURL: process.env.NEXT_PUBLIC_SEGMENT_CDN_URL,
  })
```

I would not like to teach people to do this ... it will cause issues like caching the wrong value or somebody blocking the `localStorage` ... you named it.

You do you, but I rather teach people to put such information in the `env` file since that is probably where people should be injecting such thing, and it will fix your issue

Reply:
- I didn't even consider that people might want to override cdnURL, but that's a good idea!
- on the subject of writeKey -- on a regular application, I agree. However, this is a playground for debugging (run `make dev` to see) -- so there's a writeKey input field that allows people to swap out writeKeys at runtime.

Reply:
> I didn't even consider that people might want to override cdnURL, but that's a good idea!

Well, we have to do a proxy to Segment to bypass AdBlocker ... so probably most companies that know a bit about infra would do that most of the time 😄

Reply:
> run make dev if you haven't yet to see what I mean!

I see I see, would call the directory "playground" because people are getting used to the idea of using `examples` directory to learn how to get up to speed with this ... which is my actually worry (not you, but others reading code out of context)

Reply:
> > I didn't even consider that people might want to override cdnURL, but that's a good idea!
> 
> Well, we have to do a proxy to Segment to bypass AdBlocker ... so probably most companies that know a bit about infra would do that most of the time 😄

yep, I just meant the few of us who use the playground — I’d just seen it used to demo segment and develop locally directly against segment itself. Anyway, now it’s overridavle!

Reply:
> > run make dev if you haven't yet to see what I mean!
> 
> I see I see, would call the directory "playground" because people are getting used to the idea of using `examples` directory to learn how to get up to speed with this ... which is my actually worry (not you, but others reading code out of context)

I think it’s meant to serve a duel role, but that’s a good point — it’s not typically what you think of as an ‘example’ in an examples repo!